x-pack/plugins/cases/public/components/markdown_editor/use_plugins.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  getDefaultEuiMarkdownParsingPlugins,
+  getDefaultEuiMarkdownProcessingPlugins,
+  getDefaultEuiMarkdownUiPlugins,
+} from '@elastic/eui';
+import { useMemo } from 'react';
+import { useTimelineContext } from '../timeline_context/use_timeline_context';
+import { TemporaryProcessingPluginsType } from './types';
+
+export const usePlugins = () => {
+  const timelinePlugins = useTimelineContext()?.editor_plugins;
+
+  return useMemo(() => {
```

Comment:
wondering if we should use `useMemo` here or not

Reply:
I can get rid of the `useMemo`, I'd just need to move all the `getDefault*` references out of the hook so a new reference isn't created every single time the hook is called because that caused an infinite loop to happen before 😂. I initially wanted to avoid mutating them as singleton's to guard against any weird bugs that may come up. For example, if someone was to navigate from the cases in one app instance that does have timeline enabled directly to cases in another app instance that doesn't have it enabled. New plugin instances would probably be created in that scenario depending on how navigation happens, but wanted to avoid potential bugs there.

Reply:
then leave it, i dont want a bug! lol